### PR TITLE
Migrate names to conform KubeVirt conventions

### LINKED ISSUE #5: [TASK] Migrate various names to conform KubeVirt conventions
Due to the upcoming [project migration schedule](https://github.com/kubevirt/community/blob/main/design-proposals/kubevirtbmc.md), we have some tasks that need to be done before the migration to ensure smooth progress.

One of the tasks is to renaming the CRD and change the API group of it to conform the KubeVirt convention:

- Changing from `KubeBMC` to `VirtualMachineBMC`
- Changing from `zespre.com` to `kubevirt.org`

Also, the components `kubebmc-controller` and `kbmc` should be renamed to `virt-bmc-controller` and `virt-bmc`, respectively.

Finally, the project's name will be changed to `KubeVirtBMC` to convey that this project is targeted to work with KubeVirt.

P.S. This is a follow-up issue of #1.

### PR DESCRIPTION
**Pull request by kashifest**
Fixes #5

### CODE REVIEW

**Comment by kashifest** [comment]
@starbops please review, theres so many changes that I might have easily overlooked something.

**Comment by kashifest** [comment]
@starbops did you get a chance to look into this ?

**Comment by starbops** [comment]
@kashifest I just came back from vacation. Working on it 💪

File: internal/controller/suite_test.go [hunk 1]
```diff
@@ -34,10 +34,10 @@ import (
...
 	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 
-	virtualmachinev1 "zespre.com/kubebmc/api/v1"
-	"zespre.com/kubebmc/internal/controller/kubebmc"
-	"zespre.com/kubebmc/internal/controller/service"
-	"zespre.com/kubebmc/internal/controller/virtualmachine"
+	virtualmachinev1 "kubevirt.org/virtualmachinebmc/api/v1"
+	"kubevirt.org/virtualmachinebmc/internal/controller/service"
+	"kubevirt.org/virtualmachinebmc/internal/controller/virtualmachine"
+	"kubevirt.org/virtualmachinebmc/internal/controller/virtualmachinebmc"
```
Comment (starbops) [inline internal/controller/suite_test.go:40]:
```suggestion
	virtualmachinev1 "kubevirt.org/kubevirtbmc/api/v1"
	"kubevirt.org/kubevirtbmc/internal/controller/service"
	"kubevirt.org/kubevirtbmc/internal/controller/virtualmachine"
	"kubevirt.org/kubevirtbmc/internal/controller/virtualmachinebmc"
```

File: internal/controller/virtualmachine/controller.go [hunk 2]
```diff
@@ -25,16 +25,16 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
 	kubevirtv1 "kubevirt.io/api/core/v1"
+	virtualmachinev1 "kubevirt.org/virtualmachinebmc/api/v1"
+	ctlvirtualmachinebmc "kubevirt.org/virtualmachinebmc/internal/controller/virtualmachinebmc"
```
Comment (starbops) [inline internal/controller/virtualmachine/controller.go:29]:
```suggestion
	virtualmachinev1 "kubevirt.org/kubevirtbmc/api/v1"
	ctlvirtualmachinebmc "kubevirt.org/kubevirtbmc/internal/controller/virtualmachinebmc"
```

File: internal/controller/virtualmachinebmc/constant.go [hunk 3]
```diff
@@ -10,7 +10,7 @@ const (
 	ipmiSvcPort       = 623
 	ipmiPortName      = "ipmi"
 	ManagedByLabel    = "app.kubernetes.io/managed-by"
-	KBMCNameLabel     = "zespre.com/kubebmc-name"
-	VMNameLabel       = "zespre.com/vm-name"
-	KBMCNamespace     = "kubebmc-system"
+	KBMCNameLabel     = "kubevirt.org/virtualmachinebmc-name"
+	VMNameLabel       = "kubevirt.org/vm-name"
+	KBMCNamespace     = "virtualmachinebmc-system"
```
Comment (starbops) [inline internal/controller/virtualmachinebmc/constant.go:15]:
```suggestion
	KBMCNamespace     = "kubevirtbmc-system"
```

File: internal/controller/virtualmachinebmc/controller.go [hunk 4]
```diff
@@ -29,11 +29,11 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/log"
 
-	virtualmachinev1 "zespre.com/kubebmc/api/v1"
+	virtualmachinev1 "kubevirt.org/virtualmachinebmc/api/v1"
```
Comment (starbops) [inline internal/controller/virtualmachinebmc/controller.go:32]:
```suggestion
	virtualmachinev1 "kubevirt.org/kubevirtbmc/api/v1"
```

File: internal/controller/virtualmachinebmc/controller.go [hunk 5]
```diff
@@ -79,14 +79,14 @@ func (r *KubeBMCReconciler) constructPodForKubeBMC(kubeBMC *virtualmachinev1.Kub
 					},
 				},
 			},
-			ServiceAccountName: "kubebmc-kbmc",
+			ServiceAccountName: "virtualmachinebmc-kbmc",
```
Comment (starbops) [inline internal/controller/virtualmachinebmc/controller.go:82]:
```suggestion
			ServiceAccountName: "kubevirtbmc-virt-bmc",
```

**Review by starbops — COMMENTED** [review]
I'm sorry I didn't mention it clearly in the issue description: the project name will be changed to `KubeVirtBMC`, and the CRD name instead will be changed to `VirtualMachineBMC`. The original name `KubeBMC` was used in both the project name and the CRD name, which caused confusion. So we're going to separate them. For the component names, the controller manager will be named `virt-bmc-controller`, and the virtual BMC itself will be named `virt-bmc`. This is to follow better the `virt-*` naming convention of the KubeVirt project, such as `virt-handler` and `virt-launcher`.

I'll update the remaining variable names that are inside the control loops. No worries. Thank you for the work!

**Comment by kashifest** [comment]
@starbops PTAL again, I hope I have addressed all the comments correctly

File: internal/controller/service/controller.go [hunk 6]
```diff
@@ -57,33 +55,33 @@ func (s *ServiceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
...
 		return ctrl.Result{}, nil
 	}
 	knn := types.NamespacedName{
 		Namespace: req.Namespace,
-		Name:      kBMCName,
+		Name:      virt - bmcName,
 	}
 
-	var kubeBMC virtualmachinev1.KubeBMC
+	var kubeBMC virtualmachinev1.VirtualMachineBMC
```
Comment (starbops) [inline internal/controller/service/controller.go:67]:
```suggestion
	var virtualMachineBMC virtualmachinev1.VirtualMachineBMC
```

File: internal/controller/service/controller.go [hunk 7]
```diff
@@ -57,33 +55,33 @@ func (s *ServiceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
...
 	kubeBMC.Status.Ready = true
 	kubeBMC.Status.ServiceIP = svc.Spec.ClusterIP
 	if err := s.Status().Update(ctx, &kubeBMC); err != nil {
-		log.Error(err, "unable to update KubeBMC status")
+		log.Error(err, "unable to update VirtualMachineBMC status")
 		return ctrl.Result{}, err
 	}
 
-	log.V(1).Info("updated KubeBMC status for Service", "kubeBMC", kubeBMC)
+	log.V(1).Info("updated VirtualMachineBMC status for Service", "kubeBMC", kubeBMC)
```
Comment (starbops) [inline internal/controller/service/controller.go:84]: `kubeBMC` -> `virtualMachineBMC`

File: internal/controller/virtualmachine/controller.go [hunk 8]
```diff
@@ -108,37 +108,37 @@ func (v *VirtualMachineReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, err
 	}
 
-	// Prepare the KubeBMC
-	kubeBMC, err := v.constructKubeBMCForVirtualMachine(&vm)
+	// Prepare the VirtualMachineBMC
+	kubeBMC, err := v.constructVirtualMachineBMCForVirtualMachine(&vm)
```
Comment (starbops) [inline internal/controller/virtualmachine/controller.go:112]:
```suggestion
	toCreate, err := v.constructVirtualMachineBMCForVirtualMachine(&vm)
```

File: internal/controller/virtualmachine/controller.go [hunk 9]
```diff
@@ -108,37 +108,37 @@ func (v *VirtualMachineReconciler) Reconcile(ctx context.Context, req ctrl.Reque
...
-		log.Error(err, "unable to construct kubebmc from template")
+		log.Error(err, "unable to construct virtualmachinebmc from template")
 		return ctrl.Result{}, err
 	}
 
-	// Create the KubeBMC on the cluster
+	// Create the VirtualMachineBMC on the cluster
 	if err := v.Create(ctx, kubeBMC); err != nil && !apierrors.IsAlreadyExists(err) {
-		log.Error(err, "unable to create KubeBMC for VirtualMachine", "kubeBMC", kubeBMC)
+		log.Error(err, "unable to create VirtualMachineBMC for VirtualMachine", "kubeBMC", kubeBMC)
```
Comment (starbops) [inline internal/controller/virtualmachine/controller.go:120]:
```suggestion
		log.Error(err, "unable to create VirtualMachineBMC for VirtualMachine", "virtualMachineBMC", toCreate)
```

File: internal/controller/virtualmachine/controller.go [hunk 10]
```diff
@@ -108,37 +108,37 @@ func (v *VirtualMachineReconciler) Reconcile(ctx context.Context, req ctrl.Reque
...
 		return ctrl.Result{}, err
 	}
 
-	log.V(1).Info("created KubeBMC for VirtualMachine", "kubeBMC", kubeBMC)
+	log.V(1).Info("created VirtualMachineBMC for VirtualMachine", "kubeBMC", kubeBMC)
```
Comment (starbops) [inline internal/controller/virtualmachine/controller.go:124]:
```suggestion
	log.V(1).Info("created VirtualMachineBMC for VirtualMachine", "virtualMachineBMC", toCreate)
```

File: internal/controller/virtualmachine/controller.go [hunk 11]
```diff
@@ -108,37 +108,37 @@ func (v *VirtualMachineReconciler) Reconcile(ctx context.Context, req ctrl.Reque
...
+	kubeBMC, err := v.constructVirtualMachineBMCForVirtualMachine(&vm)
 	if err != nil {
-		log.Error(err, "unable to construct kubebmc from template")
+		log.Error(err, "unable to construct virtualmachinebmc from template")
 		return ctrl.Result{}, err
 	}
 
-	// Create the KubeBMC on the cluster
+	// Create the VirtualMachineBMC on the cluster
 	if err := v.Create(ctx, kubeBMC); err != nil && !apierrors.IsAlreadyExists(err) {
```
Comment (starbops) [inline internal/controller/virtualmachine/controller.go:119]:
```suggestion
	if err := v.Create(ctx, toCreate); err != nil && !apierrors.IsAlreadyExists(err) {
```

File: internal/controller/virtualmachine/controller.go [hunk 12]
```diff
@@ -108,37 +108,37 @@ func (v *VirtualMachineReconciler) Reconcile(ctx context.Context, req ctrl.Reque
...
-		Namespace: ctlkubebmc.KBMCNamespace,
+		Namespace: ctlvirtualmachinebmc.KBMCNamespace,
 		Name:      fmt.Sprintf("%s-%s", vm.Namespace, vm.Name),
 	}
-	var kbmc virtualmachinev1.KubeBMC
-	if err := v.Get(ctx, knn, &kbmc); err != nil {
-		log.Error(err, "unable to fetch KubeBMC")
+	var virt-bmc virtualmachinev1.VirtualMachineBMC
+	if err := v.Get(ctx, knn, &virt-bmc); err != nil {
+		log.Error(err, "unable to fetch VirtualMachineBMC")
```
Comment (starbops) [inline internal/controller/virtualmachine/controller.go:132]:
```suggestion
	var virtualMachineBMC virtualmachinev1.VirtualMachineBMC
	if err := v.Get(ctx, knn, &virtualMachineBMC); err != nil {
		log.Error(err, "unable to fetch VirtualMachineBMC")
```

File: internal/controller/virtualmachine/controller.go [hunk 13]
```diff
@@ -108,37 +108,37 @@ func (v *VirtualMachineReconciler) Reconcile(ctx context.Context, req ctrl.Reque
...
 		return ctrl.Result{}, err
 	}
 
 	if !vm.ObjectMeta.DeletionTimestamp.IsZero() {
-		if err := v.Delete(ctx, &kbmc); err != nil {
-			log.Error(err, "unable to delete KubeBMC for VirtualMachine", "kbmc", kbmc)
+		if err := v.Delete(ctx, &virt-bmc); err != nil {
+			log.Error(err, "unable to delete VirtualMachineBMC for VirtualMachine", "virt-bmc", virt-bmc)
```
Comment (starbops) [inline internal/controller/virtualmachine/controller.go:138]:
```suggestion
		if err := v.Delete(ctx, &virtualMachineBMC); err != nil {
			log.Error(err, "unable to delete VirtualMachineBMC for VirtualMachine", "virtualMachineBMC", virtualMachineBMC)
```

File: internal/controller/virtualmachine/controller.go [hunk 14]
```diff
@@ -108,37 +108,37 @@ func (v *VirtualMachineReconciler) Reconcile(ctx context.Context, req ctrl.Reque
...
 			return ctrl.Result{}, err
 		}
-		log.V(1).Info("removed KubeBMC for VirtualMachine", "kbmc", kbmc)
+		log.V(1).Info("removed VirtualMachineBMC for VirtualMachine", "virt-bmc", virt-bmc)
```
Comment (starbops) [inline internal/controller/virtualmachine/controller.go:141]:
```suggestion
		log.V(1).Info("removed VirtualMachineBMC for VirtualMachine", "virtualMachineBMC", virtualMachineBMC)
```

File: pkg/bmc/ipmi.go
Comment (starbops) [inline pkg/bmc/ipmi.go:1]: Please change the dir name to `virtbmc`

File: pkg/bmc/k8s.go
Comment (starbops) [inline pkg/bmc/k8s.go:1]: Please change the dir name to `virtbmc`

File: pkg/bmc/kbmc.go
Comment (starbops) [inline pkg/bmc/kbmc.go:1]: Please change the dir & file name to `virtbmc` & `virtbmc.go`

File: internal/controller/virtualmachinebmc/controller.go [hunk 15]
```diff
@@ -119,35 +119,35 @@ func (r *KubeBMCReconciler) constructServiceForKubeBMC(kubeBMC *virtualmachinev1
...
 // the user.
 //
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.16.3/pkg/reconcile
-func (r *KubeBMCReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+func (r *VirtualMachineBMCReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	log := log.FromContext(ctx)
 
-	var kubeBMC virtualmachinev1.KubeBMC
+	var kubeBMC virtualmachinev1.VirtualMachineBMC
```
Comment (starbops) [inline internal/controller/virtualmachinebmc/controller.go:140]:
```suggestion
	var virtualMachineBMC virtualmachinev1.VirtualMachineBMC
```

File: internal/controller/virtualmachinebmc/controller.go [hunk 16]
```diff
@@ -119,35 +119,35 @@ func (r *KubeBMCReconciler) constructServiceForKubeBMC(kubeBMC *virtualmachinev1
...
 	if err := r.Get(ctx, req.NamespacedName, &kubeBMC); err != nil {
```
Comment (starbops) [inline internal/controller/virtualmachinebmc/controller.go:141]:
```suggestion
	if err := r.Get(ctx, req.NamespacedName, &virtualMachineBMC); err != nil {
```

File: internal/controller/virtualmachinebmc/controller.go [hunk 17]
```diff
@@ -119,35 +119,35 @@ func (r *KubeBMCReconciler) constructServiceForKubeBMC(kubeBMC *virtualmachinev1
...
 		}
-		log.Error(err, "unable to fetch KubeBMC")
+		log.Error(err, "unable to fetch VirtualMachineBMC")
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	// Prepare the kbmc Pod
-	pod, err := r.constructPodForKubeBMC(&kubeBMC)
+	// Prepare the virt-bmc Pod
+	pod, err := r.constructPodForVirtualMachineBMC(&kubeBMC)
```
Comment (starbops) [inline internal/controller/virtualmachinebmc/controller.go:150]:
```suggestion
	pod, err := r.constructPodForVirtualMachineBMC(&virtualMachineBMC)
```

File: internal/controller/virtualmachinebmc/controller.go [hunk 18]
```diff
@@ -156,16 +156,16 @@ func (r *KubeBMCReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
...
 		return ctrl.Result{}, err
 	}
 
-	log.V(1).Info("created Pod for KubeBMC", "pod", pod)
+	log.V(1).Info("created Pod for VirtualMachineBMC", "pod", pod)
 
-	// Prepare the kbmc Service
-	svc, err := r.constructServiceForKubeBMC(&kubeBMC)
+	// Prepare the virt-bmc Service
+	svc, err := r.constructServiceForVirtualMachineBMC(&kubeBMC)
```
Comment (starbops) [inline internal/controller/virtualmachinebmc/controller.go:168]:
```suggestion
	svc, err := r.constructServiceForVirtualMachineBMC(&virtualMachineBMC)
```

**Review by starbops — COMMENTED** [review]
Please make sure at least the project is still buildable. I'm working on the CI stuff to ensure this is done automatically, but for now, please confirm it manually. Thank you.

**Comment by starbops** [comment]
Hi @kashifest

If you don't mind, I can help fix those bits. I know it's kind of tedious ;-)
Thank you for the work, really appreciate it.

**Comment by kashifest** [comment]
Hi, @starbops please go ahead. I am in vacation so it will take some time for me to get back onto this

**Comment by starbops** [comment]
@kashifest I'm unable to continue working on this PR by pushing commits to your branch. I'll keep your work and file a new PR. Thank you!

**Pull request closed without merging**